miner/src/miner.rs

```diff
@@ -31,19 +35,29 @@ impl Miner {
         workers: &[WorkerConfig],
     ) -> Miner {
         let (seal_tx, seal_rx) = unbounded();
+        let mp = MultiProgress::new();
 
         let worker_controllers = workers
             .iter()
-            .map(|config| start_worker(Arc::clone(&pow), config, seal_tx.clone()))
+            .map(|config| start_worker(Arc::clone(&pow), config, seal_tx.clone(), &mp))
             .collect();
 
+        let pb = mp.add(ProgressBar::new(100));
+        pb.set_style(ProgressStyle::default_bar().template("{msg:.green}"));
+
+        thread::spawn(move || {
```

Comment:
Let Miner own it as a internal field like `_mp`, then we don't need to start a thread just to prevent it being drop.